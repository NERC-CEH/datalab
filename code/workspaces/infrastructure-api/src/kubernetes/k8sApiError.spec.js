import K8sApiError from './k8sApiError';

describe('K8sApiError', () => {
  const k8sErrorResponse = {
    response: {
      statusCode: 404,
      body: {
        status: 'Expected Status',
        reason: 'Expected Reason',
        message: 'Expected Message',
      },
    },
  };

  it('assigns the provided message to to the message variable', () => {
    const error = new K8sApiError(k8sErrorResponse, 'Expected error message');
    expect(error.message).toEqual('Expected error message');
  });

  it('constructs the correct error string from the provided error response', () => {
    const error = new K8sApiError(k8sErrorResponse);
    expect(error.getErrorString()).toMatchSnapshot();
  });
});
