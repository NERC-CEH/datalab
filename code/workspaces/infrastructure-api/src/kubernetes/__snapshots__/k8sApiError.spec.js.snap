// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`K8sApiError constructs the correct error string from the provided error response 1`] = `
"K8s Api request failed with:
  Status - Expected Status
  Code - 404
  Reason - Expected Reason
  Message - Expected Message"
`;
